assets/app/scripts/services/data.js

```diff
@@ -641,6 +640,9 @@ angular.module('openshiftConsole')
         return;
       }
 
+      // Reset the retries count once we've received a message with no error.
+      this._watchWebsocketRetries(type, context, 0);
```

Comment:
also, comment why the reset is placed where it is (after resourceVersion to avoid looping, but before callbacks, to avoid falsely flagging server connection errors because of callback errors)